cli/mender-artifact/copy_test.go

```diff
@@ -130,15 +131,36 @@ func TestCopy(t *testing.T) {
 			name: "data on stdin",
 			argv: []string{"mender-artifact", "cp", ":/etc/mender/artifact_info", "-"},
 		},
+		{
+			name: "Install: Error on no file permissions given",
+			argv: []string{"mender-artifact", "install", "testkey", ":/etc/mender/testkey.key"},
+			err:  "File permissions needs to be set, if you are simply copying, the cp command should fit your needs",
+		},
+		{
+			name: "Install file, standard permissions (0600)",
+			initfunc: func() {
+				require.Nil(t, ioutil.WriteFile("testkey", []byte("foobar"), 0644))
+			},
+			argv: []string{"mender-artifact", "install", "-m", "0600", "testkey", ":/etc/mender/testkey.key"},
+			verifyTestFunc: func(imgpath string) {
+				cmd := exec.Command(filepath.Join(dir, "mender-artifact"), "cat", imgpath+":/etc/mender/testkey.key")
+				var out bytes.Buffer
+				cmd.Stdout = &out
+				err := cmd.Run()
+				assert.Nil(t, err, "got unexpected error: %v", err)
+				assert.Equal(t, "foobar", out.String())
+				// Cleanup the testkey
+				assert.Nil(t, os.Remove("testkey"))
```

Comment:
Shouldn't you test the permissions here as well, in addition to the contents?

Reply:
yes, sir!